### ISSUE TITLE
downloaded Logs are not openable in windows

### ISSUE THREAD

**Issue by dineshr93** [eclipse-apoapsis/ort-server#1415]
commit: https://github.com/eclipse-apoapsis/ort-server/commit/d52f27f9869b1775d8a84a5f0e04c2aa2a4d95b8

I am trying to perform sample scan on random public repo https://github.com/strapi/strapi.git

I received error in analyzer

![image](https://github.com/user-attachments/assets/ab449525-65d3-44e3-9359-fe3cb2bb2d6f)

I downloaded the logs from the left bottom `Logs` option
when I open the it is throwing corrupt file
![image](https://github.com/user-attachments/assets/16997d5b-9e90-406c-8ac3-1abbbdd0eca3)

tried for other scans too it didnt work. but normal **docker logs** works but it has too many noise from all the scans

`docker logs -n 50 ort-server-analyzer-worker-1`

<details>

<summary>Error log retrieved from above docker logs command </summary>

```js
   java.io.FileNotFoundException: /tmp/ort-AnalyzerDownloader-analyzer-worker16453183605341371899/docs/.yarnrc.yml (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:213)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:152)
        at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:131)
        at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:131)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2Kt.getYarnExecutable(Yarn2.kt:679)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2Kt.access$getYarnExecutable(Yarn2.kt:1)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.command(Yarn2.kt:168)
        at org.ossreviewtoolkit.utils.common.CommandLineTool$DefaultImpls.run(CommandLineTool.kt:72)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.run(Yarn2.kt:112)
        at org.ossreviewtoolkit.utils.common.CommandLineTool$DefaultImpls.run(CommandLineTool.kt:82)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.run(Yarn2.kt:112)
        at org.ossreviewtoolkit.utils.common.CommandLineTool$DefaultImpls.getVersion(CommandLineTool.kt:88)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.getVersion(Yarn2.kt:177)
        at org.ossreviewtoolkit.utils.common.CommandLineTool$DefaultImpls.checkVersion(CommandLineTool.kt:104)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.checkVersion(Yarn2.kt:112)
        at org.ossreviewtoolkit.plugins.packagemanagers.node.yarn2.Yarn2.beforeResolution(Yarn2.kt:193)
        at org.ossreviewtoolkit.analyzer.PackageManager.resolveDependencies(PackageManager.kt:284)
        at org.ossreviewtoolkit.analyzer.PackageManagerRunner$run$3.invokeSuspend(Analyzer.kt:321)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:101)
        at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:113)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:589)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:823)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:720)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:707)
2024-11-12 15:17:35.988 [DefaultDispatcher-worker-1] [traceId=b40d78b7-5f18-4151-954c-9bb39a7dbcec, ortRunId=10, analyzerJobId=10, component=analyzer-worker] level=INFO  o.e.a.o.transport.MessagePublisher - Sending 'AnalyzerWorkerError' message to 'OrchestratorEndpoint'. TraceID: 'b40d78b7-5f18-4151-954c-9bb39a7dbcec'.
```

</details>

**Comment by sschuberth** [comment]
I'll trigger a run on your instance for that project to double-check.

**Comment by sschuberth** [comment]
Works for me on Linux, will probably test later on Windows, too. The analyzer error is

> 2024-11-12 16:36:14.473 [DefaultDispatcher-worker-2] [] level=ERROR o.o.analyzer.AnalyzerResultBuilder - Multiple projects with the same id 'NPM:@strapi:e2e-test-app:0.1.0' found. Not adding the project defined in 'https://github.com/strapi/strapi.git/tests/e2e/app-template/package.json' to the analyzer results as it duplicates the project defined in 'https://github.com/strapi/strapi.git/tests/cli/app-template/package.json'.

which is something we're aware of in upstream ORT.

**Comment by mnonnenmacher** [comment]
@dineshr93 Do you use Docker Compose? Then the issue is probably related to #168.

**Comment by dineshr93** [comment]
> @dineshr93 Do you use Docker Compose?

Yes.

> Then the issue is probably related to #168.

- I tried to get logs via API, I get the same error as of #168  
- where as In UI, I can able to retrieve/download the `*-logs.zip` from the UI. This downloaded log is just `<1 kb` in size and hence not unzip'able in both windows or linux.

> Works for me on Linux,
for me I placed the zip in linux and tried to extract and got below error.
![image](https://github.com/user-attachments/assets/a761d329-8ace-45dc-94dd-f79b2f027791)

and size is just  `161 byte`
![image](https://github.com/user-attachments/assets/34522baf-ec3d-41f6-acde-f61b0c05d582)

**Comment by mmurto** [comment]
> Looks like the UI is not handling the error correctly and writing the error response to the file. @mmurto Is it even possible to improve that with the way downloads are currently implemented or does this depend on #178?

Doesn't depend on it - [this](https://github.com/eclipse-apoapsis/ort-server/blob/2482ada55f14d42d845cbaeb99a083aecc0bde9c/ui/src/routes/_layout/organizations/%24orgId/products/%24productId/repositories/%24repoId/_layout/runs/%24runIndex/logs/index.tsx#L60-L67) does not throw when the API returns an error code as the fetch succeeds, so it should be modified to throw and display an error in the UI if the response status in not 2xx.
